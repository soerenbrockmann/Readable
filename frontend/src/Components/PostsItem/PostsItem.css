.postsitem {
  display: flex;
  flex-direction: column;
  border: 1px solid #f8f8f8;
  padding: 20px;
  background-color: #f8f8f8;
  border-bottom: 1px solid #c8c8c8;
}

.title {
  font-weight: bold;
  margin-bottom: 5px;
}

.body {
  margin-bottom: 10px;
}

.nav {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  margin-bottom: 10px;
  flex-wrap: wrap;
}

.nav > div:nth-child(3) {
  margin-left: auto;
}

.votescore {
  margin-right: 1em;
  color: #3399ff;
}

.commentcount {
  color: #3399ff;
}
