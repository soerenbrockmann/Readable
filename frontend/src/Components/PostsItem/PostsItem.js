import React from "react";
import PropTypes from "prop-types";
import "./PostsItem.css";
import FormatDate from "../FormatDate";

const PostsItem = props => {
  const { item } = props;
  const {
    timestamp,
    title,
    body,
    author,
    category,
    voteScore,
    commentCount
  } = item;

  return (
    <div className="postsitem">
      <div className="title">{title}</div>
      <div className="meta">
        <div>{author} - </div> <FormatDate timestamp={timestamp} />
      </div>
      <div className="body">{body}</div>
      <div className="nav">
        <div className="votescore">{voteScore} Votes</div>
        <div className="commentcount">{commentCount} Comments</div>
        <div className="category">[{category}]</div>
      </div>
    </div>
  );
};

FormatDate.propTypes = {
  timestamp: PropTypes.number.isRequired
};

export default PostsItem;
