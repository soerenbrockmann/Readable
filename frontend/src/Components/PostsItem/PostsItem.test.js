import React from "react";
import { shallow } from "enzyme";
import PostsItem from "./PostsItem";

describe("<PostsItem />", () => {
  const item = {
    timestamp: 1468166872634,
    title: "",
    body: "",
    author: "",
    category: "",
    voteScore: 1,
    commentCount: 2
  };
  const postsItem = shallow(<PostsItem item={item} />);
  it("renders properly", () => {
    expect(postsItem).toMatchSnapshot();
  });
});
