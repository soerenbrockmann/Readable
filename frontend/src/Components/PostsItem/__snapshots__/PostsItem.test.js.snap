// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsItem /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostsItem
    item={
        Object {
            "author": "",
            "body": "",
            "category": "",
            "commentCount": 2,
            "timestamp": 1468166872634,
            "title": "",
            "voteScore": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
/>,
        <div
          className="meta"
>
          <div>
                    
                     - 
          </div>
           
          <FormatDate
                    timestamp={1468166872634}
          />
</div>,
        <div
          className="body"
/>,
        <div
          className="nav"
>
          <div
                    className="votescore"
          >
                    1
                     Votes
          </div>
          <div
                    className="commentcount"
          >
                    2
                     Comments
          </div>
          <div
                    className="category"
          >
                    [
                    
                    ]
          </div>
</div>,
      ],
      "className": "postsitem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "title",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              
               - 
</div>,
            " ",
            <FormatDate
              timestamp={1468166872634}
/>,
          ],
          "className": "meta",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                " - ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "",
              " - ",
            ],
            "type": "div",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "timestamp": 1468166872634,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "body",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="votescore"
>
              1
               Votes
</div>,
            <div
              className="commentcount"
>
              2
               Comments
</div>,
            <div
              className="category"
>
              [
              
              ]
</div>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                1,
                " Votes",
              ],
              "className": "votescore",
            },
            "ref": null,
            "rendered": Array [
              1,
              " Votes",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                2,
                " Comments",
              ],
              "className": "commentcount",
            },
            "ref": null,
            "rendered": Array [
              2,
              " Comments",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "[",
                "",
                "]",
              ],
              "className": "category",
            },
            "ref": null,
            "rendered": Array [
              "[",
              "",
              "]",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
/>,
          <div
            className="meta"
>
            <div>
                        
                         - 
            </div>
             
            <FormatDate
                        timestamp={1468166872634}
            />
</div>,
          <div
            className="body"
/>,
          <div
            className="nav"
>
            <div
                        className="votescore"
            >
                        1
                         Votes
            </div>
            <div
                        className="commentcount"
            >
                        2
                         Comments
            </div>
            <div
                        className="category"
            >
                        [
                        
                        ]
            </div>
</div>,
        ],
        "className": "postsitem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "title",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                
                 - 
</div>,
              " ",
              <FormatDate
                timestamp={1468166872634}
/>,
            ],
            "className": "meta",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  " - ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "",
                " - ",
              ],
              "type": "div",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "timestamp": 1468166872634,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "body",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="votescore"
>
                1
                 Votes
</div>,
              <div
                className="commentcount"
>
                2
                 Comments
</div>,
              <div
                className="category"
>
                [
                
                ]
</div>,
            ],
            "className": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  1,
                  " Votes",
                ],
                "className": "votescore",
              },
              "ref": null,
              "rendered": Array [
                1,
                " Votes",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  2,
                  " Comments",
                ],
                "className": "commentcount",
              },
              "ref": null,
              "rendered": Array [
                2,
                " Comments",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "[",
                  "",
                  "]",
                ],
                "className": "category",
              },
              "ref": null,
              "rendered": Array [
                "[",
                "",
                "]",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
